pkg/proc/core/delve_core.go

```diff
@@ -0,0 +1,158 @@
+package core
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/go-delve/delve/pkg/dwarf/op"
+	"github.com/go-delve/delve/pkg/elfwriter"
+	"github.com/go-delve/delve/pkg/proc"
+)
+
+func platformFromNotes(notes []*note) (goos, goarch string, err error) {
+	for _, note := range notes {
+		if note.Type != elfwriter.DelveHeaderNoteType {
+			continue
+		}
+		lines := strings.Split(string(note.Desc.([]byte)), "\n")
+		v := strings.Split(lines[0], "/")
+		if len(v) != 2 {
+			return "", "", fmt.Errorf("malformed delve header note: %q", string(note.Desc.([]byte)))
+		}
+		return v[0], v[1], nil
+	}
+	panic("internal error")
+}
+
+func threadsFromDelveNotes(p *process, notes []*note) (proc.Thread, error) {
+	var currentThread proc.Thread
+	for _, note := range notes {
+		if note.Type == elfwriter.DelveHeaderNoteType {
+			const ()
+			for _, line := range strings.Split(string(note.Desc.([]byte)), "\n") {
```

Comment:
nit: This can be a bit cleaner / more efficient with a `bytes.Buffer` and then using the `buf.ReadString('\n')` method.